indexer/src/main/java/au/org/aodn/esindexer/service/DatasetAccessServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+package au.org.aodn.esindexer.service;
+
+import au.org.aodn.esindexer.exception.MetadataNotFoundException;
+import au.org.aodn.esindexer.model.DataRecord;
+import au.org.aodn.esindexer.model.Dataset;
+import org.springframework.http.*;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.time.LocalDate;
+import java.time.YearMonth;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DatasetAccessServiceImpl implements DatasetAccessService {
+
+    private String serverUrl;
+
+    @Override
+    public String getServiceUrl() {
+        return serverUrl;
+    }
+
+    @Override
+    public void setServiceUrl(String url) {
+        this.serverUrl = url;
+    }
+
+    public DatasetAccessServiceImpl(String serverUrl) {
+       setServiceUrl(serverUrl);
+    }
+
+    private final RestTemplate restTemplate = new RestTemplate();
+    @Override
+    public Dataset getIndexingDatasetBy(String uuid, LocalDate startDate, LocalDate endDate) {
+
+        // currently, we force to get data in the same month and year to simplify the logic
+        if (startDate.getMonth() != endDate.getMonth() || startDate.getYear() != endDate.getYear()) {
+            throw new IllegalArgumentException("Start date and end date must be in the same month and year");
+        }
+
+        try {
+            HttpEntity<String> request = getRequestEntity(null, null);
+
+            Map<String, Object> params = new HashMap<>();
+            params.put("uuid", uuid);
+
+            String url = UriComponentsBuilder.fromHttpUrl(getDatasetAccessEndpoint() + "/data/{uuid}")
+                    .queryParam("is_to_index", "true")
+                    .queryParam("start_date", startDate)
+                    .queryParam("end_date", endDate)
+                    .buildAndExpand(uuid)
+                    .toUriString();
+
+            ResponseEntity<DataRecord[]> responseEntity = restTemplate.exchange(
```

Comment:
DataRecord is a Dto? Then you should rename the model class to make it clear, with Dto postfix.

Reply:
Get....Dto, eg GetDataRecordDto

Create....Dto, e.g CreateDataRecordDto

Reply:
ah, it is not a standard DTO. Maybe i should rename `DataRecord` into `AggregatedDatum`, to express it is a single datum of a dataset with aggregated count

Reply:
or maybe just `Datum`?